src/imio/directory/core/contents/entity/export.py

```diff
@@ -0,0 +1,211 @@
+            "review_state": ["published", "private"],
+        }
+
+        # Executing the query
+        brains = api.portal.get_tool("portal_catalog")(query)
+        datas = []
+        for brain in brains:
+            obj = brain.getObject()
+            items = {}
+            for attribute in ordered_signifiant_columns:
+                try:
+                    json.dumps(getattr(obj, attribute, None))
+                except TypeError:
+                    if attribute == "modified":
+                        items[attribute] = getattr(obj, attribute)().strftime(
+                            "%Y-%m-%d %H:%M:%S.%f"
+                        )
+                    else:
+                        continue
+                else:
+                    if not attribute in items:
+                        items[attribute] = getattr(obj, attribute, None)
+                if isinstance(getattr(obj, attribute, None), dict):
+                    dict_obj = getattr(obj, attribute, None)
+                    for k, v in dict_obj.items():
+                        for v, v_value in v.items():
+                            items[f"schedule.{k}.{v}"] = v_value
+                if (
+                    attribute == "geolocation.latitude"
+                    or attribute == "geolocation.longitude"
+                ):
+                    attributes = attribute.split(".")
+                    first_attr = getattr(obj, attributes[0], None)
+                    items[f"{attributes[0]}.{attributes[1]}"] = (
+                        None
+                        if first_attr is None
+                        else getattr(first_attr, attributes[1], None)
+                    )
+                if attribute == "taxonomy_contact_category":
+                    items["taxonomy_contact_category"] = self.get_taxonomy_label_by_id(
+                        obj
+                    )
+                if attribute == "iam" and getattr(obj, attribute, None) is not None:
+                    items["iam"] = self.get_vocabulary_label(
+                        "imio.smartweb.vocabulary.IAm", items["iam"]
+                    )
+                if (
+                    attribute == "facilities"
+                    and getattr(obj, attribute, None) is not None
+                ):
+                    items["facilities"] = self.get_vocabulary_label(
+                        "imio.directory.vocabulary.Facilities", items["facilities"]
+                    )
+                if attribute == "topics" and getattr(obj, attribute, None) is not None:
+                    items["topics"] = self.get_vocabulary_label(
+                        "imio.smartweb.vocabulary.Topics", items["topics"]
+                    )
+                if attribute == "type" and getattr(obj, attribute, None) is not None:
+                    items["type"] = self.get_vocabulary_label(
+                        "imio.directory.vocabulary.ContactTypes", items["type"]
+                    )
```

Comment:
You could / shoud have used Plone / products serializers : you would have had a lot less work.
Many conversions are already done / tested / optimized for us ...

Reply:
Kamoulox